### ISSUE TITLE
prepare_meta does not exit on failed upgrades

### ISSUE THREAD

**Issue by kdopen** [iotbzh/agl-manifest#4]
If an attempt to upgrade the script fails hard, the script simply continues on. This can be problematic if you are relying on something in the updated script (e.g. a new flavor such as chinook)

```
info "Error while checking for upgrade on script $SCRIPT"
rm $tempfile
return 0
```
This makes it hard to use the script reliably in automation environments (such as jenkins)

**Comment by jobol** [comment]
@kdopen should it be closed now?

**Comment by kdopen** [comment]
No, I deliberately did not include a fix to this in my pull request

* It is an unrelated change, and shouldn't be in that commit
* It is a potentially breaking change for anyone already using the script
* You might be perfectly happy with the current behavior.

I think this needs some discussion

**Comment by kdopen** [comment]
Here's another place where it happens

`wget --quiet -O $tempfile $SELF_URL || { info "Unable to check for upgrade. Remote script not accessible"; rm -f $tempfile; return 0; }`
